content/en/blog/2021/blog-mermaid.md

```diff
@@ -0,0 +1,170 @@
+ classDef cluster fill:#fff,stroke:#bbb,stroke-width:2px,color:#326ce5;
+ class ingress,service,pod1,pod2 k8s;
+ class client plain;
+ class cluster cluster;
+```
+
+---
+<br>
+
+Did you notice that some of the diagram elements are colored blue? That is accomplished using a `classDef` statement to define a class of style attributes, and a `class` statement to define the elements to apply that class to.
+
+In the code for the diagram above, we have examples of both: 
+```
+classDef k8s fill:#326ce5,stroke:#fff,stroke-width:4px,color:#fff; // defines style for the k8s class
+class ingress,service,pod1,pod2 k8s; // k8s class is applied to elements ingress, service, pod1 and pod2.
+```
+
+You can even style an individual element in your diagram using the style statement.
+```mermaid
+style box1 fill:#FF0000,stroke:333,stroke-width:3px // applies style to box1
+```
+
+---
+<br>
+
+The next diagram appears in [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/#example-one-topologyspreadconstraint).
+
+![pod-spread](/img/pod-topo-constraint-02.svg)
+
+<p style="text-align: center; font-weight: bold"><a href="https://mermaid-js.github.io/mermaid-live-editor/edit/#eyJjb2RlIjoiZ3JhcGggIEJUXG4gICAgc3ViZ3JhcGggXCJ6b25lQlwiXG4gICAgICAgIHAzKFBvZCkgLS0-IG4zKE5vZGUzKVxuICAgICAgICBuNChOb2RlNClcbiAgICBlbmRcbiAgICBzdWJncmFwaCBcInpvbmVBXCJcbiAgICAgICAgcDEoUG9kKSAtLT4gbjEoTm9kZTEpXG4gICAgICAgIHAyKFBvZCkgLS0-IG4yKE5vZGUyKVxuICAgIGVuZFxuXG4gICAgY2xhc3NEZWYgcGxhaW4gZmlsbDojZGRkLHN0cm9rZTojZmZmLHN0cm9rZS13aWR0aDo0cHgsY29sb3I6IzAwMDtcbiAgICBjbGFzc0RlZiBrOHMgZmlsbDojMzI2Y2U1LHN0cm9rZTojZmZmLHN0cm9rZS13aWR0aDo0cHgsY29sb3I6I2ZmZjtcbiAgICBjbGFzc0RlZiBjbHVzdGVyIGZpbGw6I2ZmZixzdHJva2U6I2JiYixzdHJva2Utd2lkdGg6MnB4LGNvbG9yOiMzMjZjZTU7XG4gICAgY2xhc3MgbjEsbjIsbjMsbjQscDEscDIscDMgazhzO1xuICAgIGNsYXNzIHpvbmVBLHpvbmVCIGNsdXN0ZXI7XG4iLCJtZXJtYWlkIjoie1xuICBcInRoZW1lXCI6IFwiZGVmYXVsdFwiXG59IiwidXBkYXRlRWRpdG9yIjpmYWxzZSwiYXV0b1N5bmMiOnRydWUsInVwZGF0ZURpYWdyYW0iOnRydWV9">Live editor link: Pod Topology Spread Constraints</a></p>
+
+code:
+```mermaid
+graph TB
+subgraph "zoneB"
+   p3(Pod) --> n3(Node3)
+   n4(Node4)
+end
+subgraph "zoneA"
+   p1(Pod) --> n1(Node1)
+   p2(Pod) --> n2(Node2)
+end
+ 
+classDef plain fill:#ddd,stroke:#fff,stroke-width:4px,color:#000;
+classDef k8s fill:#326ce5,stroke:#fff,stroke-width:4px,color:#fff;
+classDef cluster fill:#fff,stroke:#bbb,stroke-width:2px,color:#326ce5;
+class n1,n2,n3,n4,p1,p2,p3 k8s;
+class zoneA,zoneB cluster;
+```
+
+---
+<br>
+
+A [recent talk at Kubecon](https://kccncna2021.sched.com/event/lV37/beyond-block-diagrams-different-ways-of-understanding-k8s-architecture-kim-schlesinger-digitalocean) suggested the use of sequence diagrams to describe the interactions between different components in the control plane and nodes. Here is an example illustrating the system flow starting with a `kubectl apply` command and ending with container startup on the node.
```

Comment:
https://www.youtube.com/watch?v=X40LJM0KuQ8 is a video of that talk.

Reply:
should I replace kubecon reference with YT reference?

Reply:
I think you should -- the kubecon page has a good description of the talk, but I think if I were sent there I'd just be looking for the YT link

Reply:
done. tx @nate-double-u